CWMS Data API documentation /timeseries GET endpoints.

rma-kayla opened the pull request:
Phase 1 to resolve #1117 

Updated existing documentation files:
 - Create consistent title, section, subsection .rst hierarchy.
 - Changed table format in authorization.rst to csv-table.  This format allows for much easier editing/formatting.
 
Created outline for Read the Docs CWMS Data API:
- Structured table of contents.
- Added pages for each of the 8 /timeseries endpoints for the GET method specifically.

- Initial internal linking to shared parameter definitions.
- Initial external linking to the CWMS Database Read the Docs.
- Added API reference page to show possible capabilities of using OpenAI specs to generate a high-level output of current Swagger-UI.

Implemented Gradle build task for read the docs.
Pending approval, we will move on with Phase 2 and implement proper definitions for parameters and endpoints.

rma-kayla committed:
Minor edits to existing files.
- Remove unnecessary/duplicate files

rma-kayla committed:
Implement gradle task for building docs module.
- Updating environment options

rma-kayla committed:
Implement new TOC tree and new documentation.
- Refactored existing index and toc tree design
- Incorporated css style sheet

rma-kayla committed:
Clean and refactor docs.
- Introduced new intersphinx mapping for CWMS database documentation.
- Minor refactoring in Gradle build scripts for consistency.

rma-kayla committed:
Implemented 8 timeseries GET documentation outline

rma-psmorris commented:
Spoke with Kayla on this PR. All changes in the PR were manually performed and intentional. I've asked her to amend the PR description or via comment to more thoroughly describe the work she has performed in this PR.

rma-kayla committed:
Merge remote-tracking branch 'origin/develop' into docs/timeseries-documentation

rma-kayla committed:
Minor cleanup and structure organization:
- fix title/section hierarchy consistency,
- css style edits
- toctree depth updates and reordering pages
- table width edits
- removal/edits to incorrect information

rma-kayla committed:
Removed openapi build setup due to failed PR build.
- Left api-reference page option available, but commented out requirements and extensions to avoid unnecessary set up.

rma-kayla committed:
Merge remote-tracking branch 'origin/develop' into docs/timeseries-documentation

rma-kayla commented:
this is ready to review @rma-psmorris 
Build was failing due to openapi options.

Minor cleanup and structure organization:
- fix title/section hierarchy consistency,
- toctree depth updates and reordering pages
- table width edits
- removal/edits to incorrect information

Removed openapi build setup due to failed PR build.
- Left api-reference page option available but commented out requirements and extensions to avoid unnecessary set up.

rma-rripken committed:
Merge branch 'develop' into docs/timeseries-documentation
